concurrent/cond.go

```diff
@@ -45,8 +46,11 @@ func (cond *TimeoutCond) HasWaiters() bool {
 // Wait waits for a signal, or for the context do be done. Returns true if signaled.
 func (cond *TimeoutCond) Wait(ctx context.Context) bool {
 	cond.addWaiter()
+
+	cond.condL.RLock()
 	//copy signal in lock, avoid data race with Interrupt
 	ch := cond.signal
+	cond.condL.RUnlock()
 	//wait should unlock mutex,  if not will cause deadlock
 	cond.L.Unlock()
```

Comment:
`Interrupt()` changed from using `cond.L` to `cond.condL`; does the same change need to be made here?

Reply:
The condL is for protected cond's signal chan, L comes from out, Unlock before wait, and lock after wait. Origin Interrupt() uses the out locker, which is a mistake.